crates/core/src/syn/parser/basic/mod.rs

```diff
@@ -134,6 +138,40 @@ impl TokenValue for Uuid {
 	}
 }
 
+impl TokenValue for File {
+	fn from_token(parser: &mut Parser<'_>) -> ParseResult<Self> {
+		if !parser.settings.files_enabled {
+			bail!("Cannot use files, as the experimental files capability is not enabled", @parser.last_span);
```

Comment:
Same here, use unexpected flag or just ensure this is never called when the flag is not enabled.